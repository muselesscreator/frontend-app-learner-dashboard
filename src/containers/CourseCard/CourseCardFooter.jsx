import React from 'react';
import { Button, PageBanner, Alert } from '@edx/paragon';
import { Program } from '@edx/paragon/icons';

export const CourseCardFooter = () => (
  <Alert variant='success'>
    footer
  </Alert>
);