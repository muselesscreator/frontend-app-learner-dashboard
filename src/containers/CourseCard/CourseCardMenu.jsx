import React from 'react';
import { Dropdown, Icon, IconButton } from '@edx/paragon';
import { MoreVert } from '@edx/paragon/icons';

export const CourseCardMenu = () => (
  <Dropdown>
    <Dropdown.Toggle
      id='dropdown-toggle-with-iconbutton'
      as={IconButton}
      src={MoreVert}
      iconAs={Icon}
      variant='primary'
      alt='Actions dropdown'
    />
    <Dropdown.Menu>
      <Dropdown.Item href='#/action-1'>Unenroll</Dropdown.Item>
      <Dropdown.Item href='#/action-2'>Email Settings</Dropdown.Item>
      <Dropdown.Item href='#/action-3'>Share to Facebook</Dropdown.Item>
      <Dropdown.Item href='#/action-3'>Share to Twitter</Dropdown.Item>
    </Dropdown.Menu>
  </Dropdown>
);
