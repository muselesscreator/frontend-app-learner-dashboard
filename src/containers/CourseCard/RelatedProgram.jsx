import React from 'react';
import { Button, useToggle, ModalDialog } from '@edx/paragon';
import { Program } from '@edx/paragon/icons';

export const RelatedProgram = () => {
  const [isOpen, open, close] = useToggle(false);
  return (
    <>
      <Button variant='tertiary' iconBefore={Program} onClick={open}>
        2 Related Program
      </Button>
      <ModalDialog
        title="My dialog"
        isOpen={isOpen}
        onClose={close}
        hasCloseButton
        isFullscreenOnMobile
      >
        <ModalDialog.Header>
          <ModalDialog.Title>
            Related Programs
          </ModalDialog.Title>
        </ModalDialog.Header>

        <ModalDialog.Body>
          <p>
            I'm baby palo santo ugh celiac fashion axe. La croix lo-fi venmo whatever. Beard man braid migas single-origin coffee forage ramps. Tumeric messenger bag bicycle rights wayfarers, try-hard cronut blue bottle health goth. Sriracha tumblr cardigan, cloud bread succulents tumeric copper mug marfa semiotics woke next level organic roof party +1 try-hard.
          </p>
        </ModalDialog.Body>
      </ModalDialog>
    </>
  );
};
