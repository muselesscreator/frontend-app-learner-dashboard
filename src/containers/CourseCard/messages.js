import { StrictDict } from 'utils';

export const messages = StrictDict({
  accessExpired: {
    id: 'learner-dash.courseCard.accessExpired',
    description: 'Course access expiration date message on course card for expired access.',
    defaultMessage: 'Access expired {accessExpirationDate}',
  },
  accessExpires: {
    id: 'learner-dash.courseCard.accessExpires',
    description: 'Course access expiration date message on course card.',
    defaultMessage: 'Access expires {accessExpirationDate}',
  },
  courseEnded: {
    id: 'learner-dash.courseCard.courseEnded',
    description: 'Course ended message on course card.',
    defaultMessage: 'Course ended {endDate}',
  },
  courseEnds: {
    id: 'learner-dash.courseCard.courseEnds',
    description: 'Course ending message on course card.',
    defaultMessage: 'Course ends {endDate}',
  },
  bannerAlt: {
    id: 'learner-dash.courseCard.bannerAlt',
    description: 'Course card banner alt-text',
    defaultMessage: 'Course thumbnail',
  },
  unknownProviderName: {
    id: 'learner-dash.courseCard.unknownProviderName',
    description: 'Provider name display when name is unknown',
    defaultMessage: 'Unknown',
  },
});

export default messages;
