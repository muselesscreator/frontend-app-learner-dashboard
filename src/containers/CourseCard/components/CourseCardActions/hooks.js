import { Locked } from '@edx/paragon/icons';

import { selectors } from 'data/redux';
import { useIntl, getCardValues } from 'hooks';
import messages from './messages';

const { cardData } = selectors;

export const actionHooks = ({ courseNumber }) => {
  const { formatMessage } = useIntl();
  const data = getCardValues(courseNumber, {
    canUpgrade: cardData.canUpgrade,
    isAudit: cardData.isAudit,
    isAuditAccessExpired: cardData.isAuditAccessExpired,
    isVerified: cardData.isVerified,
    isPending: cardData.isCourseRunPending,
    isFinished: cardData.isCourseRunFinished,
  });
  let primary;
  let secondary = null;
  if (!data.isVerified) {
    secondary = {
      iconBefore: Locked,
      variant: 'outline-primary',
      disabled: !data.canUpgrade,
      children: formatMessage(messages.upgrade),
    };
  }
  if (data.isPending) {
    primary = { children: formatMessage(messages.beginCourse) };
  } else if (!data.isFinished) {
    primary = {
      children: formatMessage(messages.resume),
      disabled: data.isAudit && data.isAuditAccessExpired,
    };
  } else {
    primary = { children: formatMessage(messages.viewCourse) };
  }
  return { primary, secondary };
};

export default actionHooks;
