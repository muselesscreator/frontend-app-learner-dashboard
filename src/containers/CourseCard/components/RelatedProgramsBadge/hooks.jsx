import React from 'react';
import { useIntl } from '@edx/frontend-platform/i18n';

import { StrictDict } from 'utils';
import { selectors } from 'data/redux';
import { useCardValues } from 'hooks';

import messages from './messages';
import * as module from './hooks';

const { cardData } = selectors;

export const state = StrictDict({
  isOpen: (val) => React.useState(val), // eslint-disable-line
});

export const useRelatedProgramsBadgeData = ({ courseNumber }) => {
  const [isOpen, setIsOpen] = module.state.isOpen(false);
  const { formatMessage } = useIntl();
  const { numPrograms } = useCardValues(courseNumber, {
    numPrograms: cardData.numRelatedPrograms,
  });

  let programsMessage = '';
  if (numPrograms) {
    programsMessage = formatMessage(
      numPrograms === 1 ? messages.badgeLabelSingular : messages.badgeLabelPlural,
      { numPrograms },
    );
  }

  return {
    numPrograms,
    programsMessage,
    isOpen,
    openModal: React.useCallback(() => setIsOpen(true), [setIsOpen]),
    closeModal: React.useCallback(() => setIsOpen(false), [setIsOpen]),
  };
};

export default useRelatedProgramsBadgeData;
