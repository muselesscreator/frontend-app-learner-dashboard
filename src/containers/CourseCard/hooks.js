import { selectors } from 'data/redux';

import { useIntl, getCardValues } from 'hooks';

import * as module from './hooks';
import messages from './messages';

const { cardData } = selectors;

export const accessMessage = ({ courseNumber }) => {
  const { formatMessage, formatDate } = useIntl();

  const data = getCardValues(courseNumber, {
    accessExpirationDate: cardData.courseRunAccessExpirationDate,
    isAudit: cardData.isAudit,
    isFinished: cardData.isCourseRunFinished,
    isAuditAccessExpired: cardData.isAuditAccessExpired,
    endDate: cardData.courseRunEndDate,
  });
  if (data.isAudit) {
    return formatMessage(
      data.isAuditAccessExpired ? messages.accessExpired : messages.accessExpires,
      { accessExpirationDate: formatDate(data.accessExpirationDate) },
    );
  }
  return formatMessage(
    data.isFinished ? messages.courseEnded : messages.courseEnds,
    { endDate: formatDate(data.endDate) },
  );
};

export const cardHooks = ({ courseNumber }) => {
  const { formatMessage } = useIntl();
  const data = getCardValues(courseNumber, {
    title: cardData.courseTitle,
    bannerUrl: cardData.courseBannerUrl,
    providerName: cardData.providerName,
  });

  return {
    title: data.title,
    bannerUrl: data.bannerUrl,
    providerName: data.providerName || formatMessage(messages.unknownProviderName),
    accessMessage: module.accessMessage({ courseNumber }),
    formatMessage,
  };
};

export default cardHooks;
