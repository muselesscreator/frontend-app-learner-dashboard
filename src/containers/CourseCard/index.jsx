import React from 'react';
import PropTypes from 'prop-types';

// import PropTypes from 'prop-types';
import { Card } from '@edx/paragon';

import hooks from './hooks';

import RelatedProgramsBadge from './components/RelatedProgramsBadge';
import CourseCardMenu from './components/CourseCardMenu';
import {
  CourseBanner,
  CertificateBanner,
  EntitlementBanner,
} from './components/Banners';
import CourseCardActions from './components/CourseCardActions';
import messages from './messages';

export const CourseCard = ({ courseNumber }) => {
  const {
    title,
    bannerUrl,
    providerName,
    accessMessage,
    formatMessage,
  } = hooks({ courseNumber });
  return (
    <div className="mb-3">
      <Card orientation="horizontal">
        <Card.ImageCap
          src={bannerUrl}
          srcAlt={formatMessage(messages.bannerAlt)}
        />
        <Card.Body>
          <Card.Header
            title={title}
            actions={<CourseCardMenu courseNumber={courseNumber} />}
          />
          <Card.Section>
            {providerName} • {courseNumber} • {accessMessage}
          </Card.Section>
          <Card.Footer
            orientation="vertical"
            textElement={<RelatedProgramsBadge courseNumber={courseNumber} />}
          >
            <CourseCardActions courseNumber={courseNumber} />
          </Card.Footer>
        </Card.Body>
      </Card>
      <div className="course-card-banners">
        <CourseBanner courseNumber={courseNumber} />
        <CertificateBanner courseNumber={courseNumber} />
        <EntitlementBanner courseNumber={courseNumber} />
      </div>
    </div>
  );
};

CourseCard.propTypes = {
  courseNumber: PropTypes.string.isRequired,
};

CourseCard.defaultProps = {};

export default CourseCard;
