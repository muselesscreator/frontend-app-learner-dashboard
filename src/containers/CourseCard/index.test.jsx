import React from 'react';
import { shallow } from 'enzyme';

import CourseCard from '.';
import hooks from './hooks';

jest.mock('./hooks', () => ({
  __esModule: true,
  default: jest.fn(),
}));

jest.mock('./components/RelatedProgramsBadge', () => 'RelatedProgramsBadge');
jest.mock('./components/CourseCardMenu', () => 'CourseCardMenu');
jest.mock('./components/banners', () => ({
  CourseBanner: () => 'CourseBanner',
  CertificateBanner: () => 'CertificateBanner',
  EntitlementBanner: () => 'EntitlementBanner',
}));
jest.mock('./components/CourseCardActions', () => 'CourseCardActions');

const dataProps = {
  title: 'hooks.title',
  bannerUrl: 'hooks.bannerUrl',
  providerName: 'hooks.providerName',
  accessMessagE: 'hooks.accessMessage',
  formatMessage: jest.fn(msg => ({ formatted: msg })),
};

const courseNumber = 'test-course-number';

describe('CourseCard component', () => {
  test('snapshot', () => {
    hooks.mockReturnValueOnce(dataProps);
    expect(shallow(<CourseCard courseNumber={courseNumber} />)).toMatchSnapshot();
    expect(hooks).toHaveBeenCalledWith({ courseNumber });
  });
});
