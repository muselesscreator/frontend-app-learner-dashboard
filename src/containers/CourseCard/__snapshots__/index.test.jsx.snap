// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseCard component snapshot 1`] = `
<div
  className="mb-3"
>
  <Card
    orientation="horizontal"
  >
    <Card.ImageCap
      src="hooks.bannerUrl"
      srcAlt={
        Object {
          "formatted": Object {
            "defaultMessage": "Course thumbnail",
            "description": "Course card banner alt-text",
            "id": "learner-dash.courseCard.bannerAlt",
          },
        }
      }
    />
    <Card.Body>
      <Card.Header
        actions={
          <CourseCardMenu
            courseNumber="test-course-number"
          />
        }
        title="hooks.title"
      />
      <Card.Section>
        hooks.providerName
         • 
        test-course-number
         • 
      </Card.Section>
      <Card.Footer
        orientation="vertical"
        textElement={
          <RelatedProgramsBadge
            courseNumber="test-course-number"
          />
        }
      >
        <CourseCardActions
          courseNumber="test-course-number"
        />
      </Card.Footer>
    </Card.Body>
  </Card>
  <div
    className="course-card-banners"
  >
    <CourseBanner
      courseNumber="test-course-number"
    />
    <CertificateBanner
      courseNumber="test-course-number"
    />
    <EntitlementBanner
      courseNumber="test-course-number"
    />
  </div>
</div>
`;
